server/manager/src/main/java/org/apache/accumulo/manager/metrics/QueueMetrics.java

```diff
@@ -0,0 +1,93 @@
+
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.apache.accumulo.core.metrics.MetricsProducer;
+import org.apache.accumulo.core.spi.compaction.CompactionExecutorId;
+import org.apache.accumulo.core.util.threads.ThreadPools;
+import org.apache.accumulo.manager.compaction.queue.CompactionJobPriorityQueue;
+import org.apache.accumulo.manager.compaction.queue.CompactionJobQueues;
+
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+
+public class QueueMetrics implements MetricsProducer {
+  private static final long DEFAULT_MIN_REFRESH_DELAY = TimeUnit.SECONDS.toMillis(5);
+  private MeterRegistry meterRegistry = null;
+  private CompactionJobQueues compactionJobQueues;
+  private AtomicLong queueCount;
+
+  public QueueMetrics(CompactionJobQueues compactionJobQueues) {
+    this.compactionJobQueues = compactionJobQueues;
+    ScheduledExecutorService scheduler = ThreadPools.getServerThreadPools()
+        .createScheduledExecutorService(1, "queueMetricsPoller", false);
+    Runtime.getRuntime().addShutdownHook(new Thread(scheduler::shutdownNow));
+    ThreadPools.watchNonCriticalScheduledTask(scheduler.scheduleAtFixedRate(this::update,
+        DEFAULT_MIN_REFRESH_DELAY, DEFAULT_MIN_REFRESH_DELAY, TimeUnit.MILLISECONDS));
+  }
+
+  public void update() {
+    if (meterRegistry != null) {
+      queueCount = meterRegistry.gauge(METRICS_COMPACTOR_JOB_PRIORITY_QUEUES, new AtomicLong(0));
+    }
+    if (queueCount != null) {
+      queueCount.set(compactionJobQueues.getQueueCount());
+    }
+
+    for (Map.Entry<CompactionExecutorId,CompactionJobPriorityQueue> entry : compactionJobQueues
+        .getQueues()) {
+      String queueId = formatString(entry.getKey().toString(), true);
+      Gauge.builder(METRICS_COMPACTOR_JOB_PRIORITY_QUEUE_LENGTH, entry, e -> e.getValue().getSize())
+          .description("Length of priority queues")
+          .tags(Tags.concat(getCommonTags(), "queue_id", queueId)).register(meterRegistry);
+
+      Gauge
+          .builder(METRICS_COMPACTOR_JOB_PRIORITY_QUEUE_JOBS_QUEUED, entry,
+              e -> e.getValue().getQueuedJobs())
+          .description("Count of current queued jobs")
+          .tags(Tags.concat(getCommonTags(), "queue_id", queueId)).register(meterRegistry);
+
+      Gauge
+          .builder(METRICS_COMPACTOR_JOB_PRIORITY_QUEUE_JOBS_DEQUEUED, entry,
+              e -> e.getValue().getDequeuedJobs())
+          .description("Count of jobs dequeued")
+          .tags(Tags.concat(getCommonTags(), "queue_id", queueId)).register(meterRegistry);
+
+      Gauge
+          .builder(METRICS_COMPACTOR_JOB_PRIORITY_QUEUE_JOBS_REJECTED, entry,
+              e -> e.getValue().getRejectedJobs())
+          .description("Count of current rejected jobs")
```

Comment:
The dequeued description does not have the word `current`, should it be removed here?  Thinking they are both similar in that they continually increase.

```suggestion
          .description("Count of rejected jobs")
```

Reply:
I agree. I pulled "current" out of each description.